### DISCUSSION TITLE
Search logic: root search

### DISCUSSION THREAD

**Discussion by dwerning** [thesaurus-linguae-aegyptiae/tla-web#114] [Ideas]
Prototype:
- root search "m" via transliteration search slot => finds all roots beginning with "m"
- root search via root search slot => finds all lemmata with any root "m" (possibly different "m"-roots)
Java app, currently:
- root search "m" via transliteration search slot => _error_
- root search via root search slot => finds all roots beginning with "m"

**Comment by simondschweitzer** [comment]
Das Verhalten von tla-web ist korrekt:

1. Wenn man in das Wurzelsuchfeld "m" eingibt, werden alle Lemmata als Treffer angezeigt, die einer Wurzel "m" zugewiesen sind.
2. Eine Suche im Transkriptionsfeld mit Wortart "Wurzel" ist sinnlos, vgl. #77 Dementsprechend können hier gar keine Treffer produziert werden.

→ tla-web ist o.k.; der Prototyp macht Quatsch

**Comment by dwerning** [comment]
cf. https://github.com/JKatzwinkel/tla-webdesign-prototype/issues/47

**Comment by dwerning** [comment]
Beide Suchwege
1. separates Suchfeld "Root"
2. Suche über Lemma-Transkription & Wortart "Root" 

sind sinnvoll, da sie unterschiedliche Ergebnismengen liefern

1. Suche "sn" im separaten Suchfeld "Root"; Ergebnis: alle _**Lemmata** deren_ Root mit "sn" anfängt 
![grafik](https://user-images.githubusercontent.com/33833393/123262448-2dc81f80-d4f8-11eb-980a-58f556e8e761.png)

2. Suche "sn" über Lemma-Transkription & Wortart "Root"; Ergebnis: alle _**Roots**_ die mit "sn" beginnen [diese Suchlogik war ursprünglich nicht vorgesehen, wurde ergänzt]
![grafik](https://user-images.githubusercontent.com/33833393/123262666-6ec03400-d4f8-11eb-99fe-db59ffc23caf.png)

**Comment by dwerning** [comment]
Die aktuelle Implementation ist unintuitiv: _Suche nach "**Wurzel**" "sn" findet nicht **Wurzeln**, sondern alle **Lemmata** deren Root mit "sn"_
Ggf. könnte man auf die ursprünglich angedachte Suchlogik  (1) _Suche nach "Wurzel" "sn" findet alle **Lemmata** deren Root mit "sn"_ verzichten, da man da intuitiver via Suche (2)  _Suche nach "sn" als "Wurzel" findet alle Roots die mit "sn"_ beginnen ran kommt.

**Comment by simondschweitzer** [comment]
> Die aktuelle Implementation ist unintuitiv: _Suche nach "**Wurzel**" "sn" findet nicht **Wurzeln**, sondern alle **Lemmata** deren Root mit "sn"_
> Ggf. könnte man auf die ursprünglich angedachte Suchlogik (1) _Suche nach "Wurzel" "sn" findet alle **Lemmata** deren Root mit "sn"_ verzichten, da man da intuitiver via Suche (2) _Suche nach "sn" als "Wurzel" findet alle Roots die mit "sn"_ beginnen ran kommt.

Dieser These möchte ich entschieden widersprechen: Entscheidend ist, was als Treffer produziert wird. Als Treffer in einer Lemmasuche erwartet man Lemmata, und nicht andere Dinge, keine Texte, keine Belegstellen und eben auch keine Wurzeln. Das Suchfeld 'Wurzel' ist eine Einschränkung, genau wie die anderen Felder, bspw. 'Übersetzung'. Als Ergebnisse sollten alle Lemmata ausgegeben werden, die der Sucheinschränkung genügen, also mit der Wurzeleigenschaft XY bzw. mit der Übersetzungseigenschaft soundso. Wenn man im Bibliographiefeld nach "Wb 5" sucht, soll als Treffer ja auch nicht Wb 5 als PDF oder als Link ausgegeben werden, sondern alle Lemmata aus Wb 5. Bietet man in einer spezialisierten Suche wie hier in der Suche nach Lemmata eine unscharfe Trefferliste, also eine Mischung aus Lemmata und Wurzeln, dann ist es nicht intuitiv.

**Comment by dwerning** [comment]
Die Kontroverse fokussiert auf die Frage, wie wir anbieten können _nach Wurzeln mit "sn"_ zu suchen (separater Reiterblock "Wurzelsuche".) Thinkng about ...
